File: DSA/Shruti-Pandey/Day_5_q2.java
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedListComparison {
    
    public static String compareLists(Node head1, Node head2) {
        Node current1 = head1;
        Node current2 = head2;
        
        while (current1 != null && current2 != null) {
            if (current1.data != current2.data) {
                return "different";
            }
            current1 = current1.next;
            current2 = current2.next;
        }
    
        if (current1 == null && current2 == null) {
            return "identical";
        } else {
            return "different";
        }
    }

    public static void main(String[] args) {
        Node head1 = new Node(1);
        head1.next = new Node(2);
        head1.next.next = new Node(3);
        Node head2 = new Node(1);
        head2.next = new Node(2);
        head2.next.next = new Node(3);
        System.out.println(compareLists(head1, head2)); 
    }
}

File: DSA/Hima-Swarupa-Padala/day5q2.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def append(self, data):
        if not self.head:
            self.head = Node(data)
        else:
            current = self.head
            while current.next:
                current = current.next
            current.next = Node(data)

    def compare_lists(self, list2):
        current1 = self.head
        current2 = list2.head

        while current1 and current2:
            if current1.data != current2.data:
                return "different"
            current1 = current1.next
            current2 = current2.next

        if current1 or current2:
            return "different"

        return "identical"

l1 = LinkedList()
l1.append(1)
l1.append(2)
l1.append(3)

l2 = LinkedList()
l2.append(1)
l2.append(2)
l2.append(3)

l3 = LinkedList()
l3.append(1)
l3.append(5)
l3.append(4)

print(l1.compare_lists(l2))  
print(l1.compare_lists(l3)) 
